tools/debug-device.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright © 2024 Red Hat, Inc.
+ *
+ * Permission to use, copy, modify, distribute, and sell this software
+ * and its documentation for any purpose is hereby granted without
+ * fee, provided that the above copyright notice appear in all copies
+ * and that both that copyright notice and this permission notice
+ * appear in supporting documentation, and that the name of Red Hat
+ * not be used in advertising or publicity pertaining to distribution
+ * of the software without specific, written prior permission.  Red
+ * Hat makes no representations about the suitability of this software
+ * for any purpose.  It is provided "as is" without express or implied
+ * warranty.
+ *
+ * THE AUTHORS DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE,
+ * INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN
+ * NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY SPECIAL, INDIRECT OR
+ * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
+ * OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+ * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ * Authors:
+ *        Peter Hutterer <peter.hutterer@redhat.com>
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <glib/gi18n.h>
+#include <glib.h>
+#include "libwacom.h"
+
+static char *database_path;
+
+static GOptionEntry opts[] = {
+        {"database", 0, 0, G_OPTION_ARG_FILENAME, &database_path, N_("Path to device database"), NULL },
+	{ .long_name = NULL}
+};
+
+/* Usage: p("function_name", "return value is %d", a) */
+#define p(f_, fmt_, ...) \
+	printf("%-40s -> " fmt_ "\n", f_, __VA_ARGS__)
+
+/* Usage: func(myfunc, "return value is %d", a) */
+#define func(f_, fmt_, ...) \
+	p(#f_"()", fmt_, __VA_ARGS__)
+
+#define strfunc(f_, dev_) \
+	func(f_, "\"%s\"", f_(dev_))
+
+#define hexfunc(f_, dev_) \
+	func(f_, "0x%x", f_(dev_))
```

Comment:
We can make it always 4 digits to be consistent with other tools (like libinput record):
```suggestion
	func(f_, "0x%04x", f_(dev_))
```